apps/crn-server/test/integration/contentful/users.integration-test.ts

```diff
@@ -0,0 +1,172 @@
+import Chance from 'chance';
+import supertest from 'supertest';
+import { Express } from 'express';
+import { v4 as uuid } from 'uuid';
+
+import { addLocaleToFields, getLinkEntity } from '@asap-hub/contentful';
+import { UserResponse } from '@asap-hub/model';
+
+import { appFactory } from '../../../src/app';
+import { getEnvironment, TestEnvironment } from '../../helpers/contentful';
+import { retryable } from '../../helpers/retryable';
+
+jest.mock('../../../src/config', () => ({
+  ...jest.requireActual('../../../src/config'),
+  logLevel: 'silent',
+}));
+
+const chance = Chance();
+
+const getUser = (props = {}) => ({
+  firstName: chance.first(),
+  lastName: chance.last(),
+  email: chance.email(),
+  role: 'Guest',
+  onboarded: true,
+  ...props,
+});
+
+jest.setTimeout(120000);
+
+describe('users', () => {
+  let app: Express;
+  let environment: TestEnvironment;
+  const loggedInUserId = uuid();
+
+  beforeAll(async () => {
+    environment = await getEnvironment();
+    app = appFactory({
+      authHandler: (req, _res, next) => {
+        req.loggedInUser = {
+          onboarded: true,
+          id: loggedInUserId,
+        } as UserResponse;
+        next();
+      },
+    });
+    const loggedIn = await environment.createEntryWithId(
+      'users',
+      loggedInUserId,
+      {
+        fields: addLocaleToFields(getUser()),
+      },
+    );
+    await loggedIn.publish();
+  });
+
+  afterAll(async () => {
+    await environment.teardown();
+  });
+
+  test('can fetch a list of users', async () => {
+    await retryable(async () => {
+      const response = await supertest(app).get('/users').expect(200);
+      expect(response.body.total).toEqual(expect.any(Number));
+      expect(response.body.items).toEqual(expect.any(Array));
+    });
+  });
+
+  test('can fetch user', async () => {
+    const fields = getUser();
+    const user = await environment.createEntry('users', {
+      fields: addLocaleToFields(fields),
+    });
+    await user.publish();
+    await retryable(async () => {
+      const response = await supertest(app)
+        .get(`/users/${user.sys.id}`)
+        .expect(200);
+      expect(response.body).toMatchObject(fields);
+    });
+  });
+
+  test('cannot fetch user if they are not onboarded', async () => {
+    const fields = getUser({
+      onboarded: false,
+    });
+    const user = await environment.createEntry('users', {
```

Comment:
Because I am of the firm belief that the create method on the data provider shouldn't exist. It serves no purpose in the software _except_ to create test fixtures, which can be done more simply by writing directly to Contentful.

If anything using the data provider makes changing the underlying implementation _more_ difficult because not only would the corresponding create method need to be updated for the new implementation, but also since it is included in test coverage metrics we would also end up needing to write unit tests for code which serves no purpose except test instrumentation. I know this because I did exactly that when writing the create method on the data provider, which was a mistake that should be undone IMO.

I'd be much happier removing that method from the data provider than passing the test setup through an IMO unnecessary abstraction.

Reply:
sure sure, you can actually remove it then, this explanation speaks to me so im cool with that